src/common/hooks/useShouldZeroApprove/useShouldZeroApprove.ts

```diff
@@ -0,0 +1,33 @@
+import { useApprovalStateForSpender } from 'lib/hooks/useApproval'
+import { useTradeSpenderAddress } from '../useTradeSpenderAddress'
+import { Currency, CurrencyAmount } from '@uniswap/sdk-core'
+import { useEffect, useState } from 'react'
+import { useTokenContract } from 'legacy/hooks/useContract'
+import { shouldZeroApprove as shouldZeroApproveFn } from './shouldZeroApprove'
+
+// TODO: Handle tokens that don't allow approvals larger than the balance of the wallet
+export function useShouldZeroApprove(amountToApprove: CurrencyAmount<Currency> | undefined): boolean {
+  const [shouldZeroApprove, setShouldZeroApprove] = useState(false)
+  const spender = useTradeSpenderAddress()
+  const tokenContract = useTokenContract(
+    amountToApprove && amountToApprove.currency.isToken ? amountToApprove.currency.address : undefined
+  )
+  const approvalState = useApprovalStateForSpender(amountToApprove, spender, () => false) // ignore approval pending state
+
+  useEffect(() => {
+    let isStale = false
+    ;(async () => {
```

Comment:
Why this?

Reply:
Running an async function inside, so if this effect somehow rerenders, we need to deal with the previous async function wanting to update state.

Reply:
But, why use this notation?
Any reason why declaring the async fn inside the `useEffect` and calling it doesn't work and it's needed to do this way?

Just asking because this notation is not that clear and I rather use it only if really needed.